tools.py

```diff
@@ -0,0 +1,284 @@
+        self.layout.addWidget(self.zmaxBox, 2, 3)
+
+        self.layout.addWidget(HorizontalLine(), 3, 0, 1, 6)
+
+        self.xResBox = QtWidgets.QSpinBox()
+        self.xResBox.setMaximum(1E6)
+        self.xResBox.setMinimum(0)
+        self.yResBox = QtWidgets.QSpinBox()
+        self.yResBox.setMaximum(1E6)
+        self.yResBox.setMinimum(0)
+        self.zResBox = QtWidgets.QSpinBox()
+        self.zResBox.setMaximum(1E6)
+        self.zResBox.setMinimum(0)
+
+        self.layout.addWidget(QtWidgets.QLabel("X steps:"), 4, 0)
+        self.layout.addWidget(self.xResBox, 4, 1)
+        self.layout.addWidget(QtWidgets.QLabel("Y steps:"), 4, 2)
+        self.layout.addWidget(self.yResBox, 4, 3)
+        self.layout.addWidget(QtWidgets.QLabel("Z steps:"), 4, 4)
+        self.layout.addWidget(self.zResBox, 4, 5)
+
+        self.layout.addWidget(HorizontalLine(), 5, 0, 1, 6)
+
+        self.dataLabelField = QtWidgets.QLineEdit()
+        self.dataLabelField.setText("Tally {}".format(cv.selectedTally))
+        self.layout.addWidget(QtWidgets.QLabel("VTK Data Label:"), 6, 0)
+        self.layout.addWidget(self.dataLabelField, 6, 1, 1, 2)
+
+        self.geomCheckBox = QtWidgets.QCheckBox()
+        self.layout.addWidget(QtWidgets.QLabel("Include Model Domains:"),
+                              7, 0, 1, 2)
+        self.layout.addWidget(self.geomCheckBox, 7, 2)
+
+        self.layout.addWidget(HorizontalLine(), 8, 0, 1, 6)
+
+        self.exportButton = QtWidgets.QPushButton("Export to VTK")
+        self.exportButton.clicked.connect(self.export_data)
+
+        self.layout.addWidget(self.exportButton, 9, 5, 1, 2)
+
+        if tally.contains_filter(openmc.MeshFilter):
+
+            mesh_filter = tally.find_filter(openmc.MeshFilter)
+            mesh = mesh_filter.mesh
+            assert(mesh.n_dimension == 3)
+
+            llc = mesh.lower_left
+            self.xminBox.setValue(llc[0])
+            self.yminBox.setValue(llc[1])
+            self.zminBox.setValue(llc[2])
+
+            urc = mesh.upper_right
+            self.xmaxBox.setValue(urc[0])
+            self.ymaxBox.setValue(urc[1])
+            self.zmaxBox.setValue(urc[2])
+
+            dims = mesh.dimension
+            self.xResBox.setValue(dims[0])
+            self.yResBox.setValue(dims[1])
+            self.zResBox.setValue(dims[2])
+
+            bounds_msg = "Using MeshFilter to set bounds automatically."
+            self.xminBox.setEnabled(False)
+            self.xminBox.setToolTip(bounds_msg)
+            self.xmaxBox.setEnabled(False)
+            self.xmaxBox.setToolTip(bounds_msg)
+            self.yminBox.setEnabled(False)
+            self.yminBox.setToolTip(bounds_msg)
+            self.ymaxBox.setEnabled(False)
+            self.ymaxBox.setToolTip(bounds_msg)
+            self.zminBox.setEnabled(False)
+            self.zminBox.setToolTip(bounds_msg)
+            self.zmaxBox.setEnabled(False)
+            self.zmaxBox.setToolTip(bounds_msg)
```

Comment:
```
for box in ('xminBox', 'xmaxBox', 'yminBox', 'ymaxBox', 'zminBox', 'zmaxBox'):
    getattr(self, box).setEnabled(False)
    getattr(self, box).setToolTip(bounds_msg)
```
Less repetitive, but maybe less clear? I'll leave it up to you if you want to adopt this.